### ISSUE TITLE
Signals refactor

### ISSUE THREAD

**Issue by totte174** [aidotse/LeakPro#306]
# Feature
Signals refactor

Refactor signal.py and attacks such that:
- [ ] Getting raw model outputs can be abstracted into the Signal base class, individual signal will simply do some numpy operations on raw model outputs.
- [ ] Allow user to enable caching of model outputs, if for example running attack with different signals. 
- [ ] Allow user to enable caching of signals in cases such as RMIA hyper-parameter tuning.
- [ ] Softmax logits in RMIA (with temperature parameter) should be a signal in signal.py
- [ ] Allow user to specify signal in attack config.
